### PR TITLE
clear credentials when configuring

### PR DESCRIPTION
**Pull request by kajes**
Clear credentials when running `sdpctl configure`. This will prevent any old tokens that may present to not be re-used before the user has logged in after re-configuring. It will also force the user to re-enter user credentials before any other command requiring the user to be signed in can be run.

Fixes: SA-19900

### CODE REVIEW

**Comment by kajes** [comment]
E2E: https://got-testrail.agi.appgate.com/index.php?/plans/view/1048643

File: pkg/configuration/configuration.go [hunk 1]
```diff
@@ -168,6 +170,19 @@ func (c *Config) ClearCredentials() error {
...
+		if util.InSlice(k, allKeys) {
+			viper.Set(k, "")
+		}
+	}
+	if err := viper.WriteConfig(); err != nil {
+		// only return error if there is a config file to write to
+		if !errors.As(err, &viper.ConfigFileNotFoundError{}) {
+			return err
+		}
+	}
```
Comment (danijeel) [inline pkg/configuration/configuration.go:185]: why is this needed? Can you prepend a inline comment why this is needed

Comment (kajes) [reply pkg/configuration/configuration.go:185]: You mean the file not found check? Well, here's what happens if you don't check for it:

Comment (danijeel) [reply pkg/configuration/configuration.go:185]: I just meant add a inline code comment in this function explaining why we need this logic, so our future self does not get confused about it (speaking mainly for myself here)

Comment (kajes) [reply pkg/configuration/configuration.go:185]: Sure, I'll do that

**Review by danijeel — APPROVED** [review]
